federatedscope/core/auxiliaries/utils.py

```diff
@@ -412,3 +413,40 @@ def format_dict(target_dict):
     elif isinstance(hooks_set, dict):
         print_obj = format_dict(hooks_set)
     return json.dumps(print_obj, indent=2).replace('\n', '\n\t')
+
+
+def get_device_info(filename):
+    if filename is None or not os.path.exists(filename):
+        logger.info('The device information file is not provided')
+        print(filename)
```

Comment:
is it necessary?

Reply:
It is used for testing and I have removed it